### ISSUE TITLE
False positives when triggering property is overriden in declaration-block-no-ignored-properties

### ISSUE THREAD

**Issue by kevinSuttle** [stylelint/stylelint#1790]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Without `declaration-block-no-duplicate-properties`, `declaration-block-no-ignored-properties`  triggers because I set a `height` and `width` after a duplicate display property.

> Which rule, if any, is this issue related to?

`declaration-block-no-ignored-properties`

> What CSS is needed to reproduce this issue?

```diff
.icon {
  border: 0;
- display: inline;
  fill: currentColor;
+  display: inline-block;
  width: 40px;
  height: 30px;
}
```

Even though that 2nd `display` declaration overrides the first, this stylelint rule still triggers because I set a `height` and `width`. 

> What stylelint configuration is needed to reproduce this issue?

```json
{
"extends": "stylelint-config-standard",
  "rules": {
    "number-leading-zero": "never",
    "custom-property-no-outside-root": true,
    "root-no-standard-properties": true,
    "selector-root-no-composition": true,
    "no-eol-whitespace": true
  }
}
```

> Which version of stylelint are you using?

`7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

What would be nice would be if stylelint saw the overridden declaration and warned me. 
I know there's this `"declaration-block-no-duplicate-properties": true,`, but I feel like it should be turned on in the `stylelint-config-standard` preset maybe. 

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged: 

```shell
 8:3  ✖  Unexpected "width" with "display: inline"    declaration-block-no-ignored-properties
 9:3  ✖  Unexpected "height" with "display: inline"   declaration-block-no-ignored-properties"
```

**Comment by jeddy3** [comment]
@kevinSuttle Thanks for creating a new issue for this and writing it up.

I think we should enhance `declaration-block-no-ignored-properties` so that the last occurrence of a property is used. I'll mark this up as help wanted. 

> `I know there's this "declaration-block-no-duplicate-properties": true`, but I feel like it should be turned on in the stylelint-config-standard preset maybe.

That's a separate issue for the `stylelint-config-standard` repo. Can you raise it there please?

**Comment by davidtheclark** [comment]
I'm not sure this is really a bug. The `display: inline` declaration is still *ignored* — it's just ignored for two different reasons in this case.

**Comment by jeddy3** [comment]
Given:

```css
a {
  display: inline;
  width: 100px;
}
```

I think it is the `width: 100px` declaration that is ignored, rather than the `display: inline` one. 

In following example the second `display` declaration sets the property to `inline-block`, and so the `width: 100px` is *no longer* ignored. However, stylelint will currently warn that it is:

```css
a {
  display: inline;
  display: inline-block;
  width: 100px;
}
```

```console
 4:2  ✖  Unexpected "width" with "display: inline"   declaration-block-no-ignored-properties"
```

It's an odd edge-case though and I'm unsure whether we should bother addressing it.

**Comment by evilebottnawi** [comment]
@jeddy3 i think be good add option `latestDecl` (need best name) for this behaviour.

**Comment by davidtheclark** [comment]
> I'm unsure whether we should bother addressing it.

I don't think it's worth addressing. I can't think of any situation in which `display: inline; display: inline-block;` is not a mistake. @evilebottnawi if you want to show it can be done in a simple PR, though, go for it.

**Comment by evilebottnawi** [comment]
@davidtheclark soon see how to solve

**Comment by jeddy3** [comment]
> if you want to show it can be done in a simple PR, though, go for it.

Agreed. @evilebottnawi I don't think an option is needed though. Just make checking the last occurrence of a property the default behaviour as it makes the rule more permissive my default.

> I can't think of any situation in which `display: inline; display: inline-block;` is not a mistake.

Same. Although, maybe there's an argument for `display: inline; display: flex` fallback situation. Either way, if @evilebottnawi can easily patch it, then fine with me.
